src/title_widget.ml

```diff
@@ -27,11 +27,11 @@ let set_title _ config soup =
     Ok ()
   | Some title_node ->
     let title_string =
-      Utils.select_any_of selectors soup >>= Soup.leaf_text |> make_title_string default_title prepend append in
+      Utils.select_any_of selectors soup >|= Soup.texts >|= String.concat " "
```

Comment:
There's [`Utils.get_element_text`](https://github.com/dmbaturin/soupault/blob/master/src/utils.ml#L110-L120) that also does some whitespace normalization. Could you check if it works fine for your test cases, and see if we should use it here?

Reply:
I don’t think this is what we want. The documentation says “Gets the first text node from an element if there's any non-empty text in it” so my intuition is that, for instance:

```
<h1>Foo <code>bar</code></h1>
```

will render

```
<title>Foo</title>
```

Which is not why I personally want. Out of curiosity, do you see a concrete issue with the proposed change?

Reply:
Oh wait, it looks like the documentation is not really aligned with the actual code. Will actually give it a try.

Reply:
Yeah, I've just noticed and fixed it. That function is used by the `toc` widget for stripping tags from headings. There's nothing wrong with your change, I just thought it may be better to reuse an existing function, so that any improvements to it automatically affect all widgets. From my quick test it handles your test case well:

```
# "<h1><code>wc</code> in language Y</h1>" |> Soup.parse |> get_element_text ;;
- : string option = Some "wc in language Y"
```